src/CoreGraphics/CGColorSpace.cs

```diff
@@ -156,10 +156,12 @@ public static CGColorSpace CreateDeviceRGB ()
 
 		public static CGColorSpace CreateCalibratedGray (nfloat [] whitepoint, nfloat [] blackpoint, nfloat gamma)
 		{
+			if (whitepoint == null)
+				throw new ArgumentNullException (nameof (whitepoint));
 			if (whitepoint.Length != 3)
-				throw new ArgumentException ("Must have 3 values", "whitepoint");
-			if (blackpoint.Length != 3)
-				throw new ArgumentException ("Must have 3 values", "blackpoint");
+				throw new ArgumentException ("Must be null or have 3 values", nameof (whitepoint));
```

Comment:
Mmh I'm confusing with `Must be null` here, I think that's a mistake given that you throw on `whitepoint == null` above.

Reply:
Also I think in all those `ArgumentException` messages you want to write: `or have exactly 3 values`, it's a `!= 3`.

Reply:
Fixed: 2f8c43e20c10acb012287c4a6bd4bae9b9d1a733